components/builder-jobsrv/src/server/scheduler.rs

```diff
@@ -262,6 +262,7 @@ impl ScheduleMgr {
 
         for group in groups {
             self.check_group(&group)?;
+            self.update_group_state(group.id as u64)?;
```

Comment:
I am not sure exactly how to call set_group_state in isolation since it is tightly coupled with the database.  
We could possibly refactor the set_group_state but given there are three areas in that function that interact with the database, not sure it is worth it either.

Reply:
I'm in favor of keeping the call to self.update_group_state to ensure we can close the group when otherwise it would not have, as in the scenario discussed in the github issue here:  https://github.com/habitat-sh/builder/issues/1714#issuecomment-1069053304

Reply:
My concern here is what happens if a group is currently being dispatched from `process_work` and then the watchdog concurrently calls `set_group_state` on that group and sets the group back to pending? Will that cause issues?

Reply:
It looks like there is 1 run thread, so the `process_work` call will have completed before `watchdog` starts, so I don't think this would cause a concurrency issue.  Then `watchdog` would process groups that are currently in Dispatching state and if it transitions back to Pending, then they should get picked up in the next iteration of the run loop.

Reply:
ahh right. yeah this should be good then.